extensions/amp-social-share/1.0/social-share.js

```diff
@@ -0,0 +1,210 @@
+        const dataStr = finalEndpoint.substr(finalEndpoint.indexOf('?'));
+        const data = parseQueryString(dataStr);
+        window.navigator.share(data).catch((e) => {
+          throwWarning(`${e.message}. ${name}`);
+        });
+      } else {
+        throwWarning(
+          `Could not complete system share.  Navigator unavailable. ${name}`
+        );
+      }
+    } else if (protocol === 'sms') {
+      openWindowDialog(
+        window,
+        finalEndpoint.replace('?', '?&'),
+        '_blank',
+        windowFeatures
+      );
+    } else {
+      openWindowDialog(window, finalEndpoint, '_blank', windowFeatures);
+    }
+  }
+
+  /**
+   * @private
+   * @param {!Event} event
+   */
+  function handleKeyPress(event) {
+    const {key} = event;
+    if (key == Keys.SPACE || key == Keys.ENTER) {
+      event.preventDefault();
+      handleActivation();
+    }
+  }
+
+  const type = props.type.toUpperCase();
+  const baseStyle = CSS.BASE_STYLE;
+  const backgroundStyle = CSS[type] || CSS.DEFAULT;
+  const size = {
+    width: checkedWidth,
+    height: checkedHeight,
+  };
+  return (
+    <div
+      role="button"
+      tabindex={props['tabIndex'] || '0'}
+      onKeyDown={handleKeyPress}
+      onClick={handleActivation}
+      style={{...size, ...props['style']}}
+      {...props}
+    >
+      <SocialShareIcon
+        style={{...backgroundStyle, ...baseStyle, ...size}}
+        type={type}
+      />
+    </div>
+  );
+}
+
+/**
+ * @param {!JsonObject} props
+ * @param {?string} name
+ * @return {!JsonObject}
+ */
+function checkProps(props, name) {
+  const {type, shareEndpoint, params, bindings, width, height} = props;
+
+  // Verify type is provided
+  if (type === undefined) {
+    throw new Error(`The type attribute is required. ${name}`);
+  }
+
+  // bindings and params props must be objects
+  if (params && !isObject(params)) {
+    throw new Error(`The params property should be an object. ${name}`);
+  }
+  if (bindings && !isObject(bindings)) {
+    throw new Error(`The bindings property should be an object. ${name}`);
+  }
+
+  // User must provide shareEndpoint if they choose a type that is not
+  // pre-configured
+  const typeConfig = getSocialConfig(type) || dict();
+  const baseEndpoint = typeConfig['shareEndpoint'] || shareEndpoint;
+  if (baseEndpoint === undefined) {
+    throw new Error(
+      `A shareEndpoint is required if not using a pre-configured type. ${name}`
+    );
+  }
+
+  // Verify width and height are valid integers
+  // Silently assigns default for undefined, null
+  // Throws Warning for booleans, strings, 0, negative numbers
+  // No errors when positive integer or equivalent string
+  let checkedWidth =
+    width === null || width === undefined ? DEFAULT_WIDTH : width;
+  let checkedHeight =
+    height === null || height === undefined ? DEFAULT_HEIGHT : height;
+  if (typeof checkedWidth === 'boolean' || !(Math.floor(checkedWidth) > 0)) {
```

Comment:
I don't understand this `if`, tbh. Why would we expect the width to be a boolean vs, say, a function or an object? What would `math.floor()` do if the width was indeed a string? IMHO, this whole warning we can do without.

Reply:
Right, the reason I have this check for boolean is because Math.floor(true) actually returns 1, which would then not trigger the warning.  Math.floor on a string converts the string to a number if it is a valid number or returns NaN which would trigger the warning.

So in combination with the above, null and undefined would silently be reassigned to be DEFAULT.  Strings, objects, booleans, functions would trigger the warning.  Valid positive strings and numbers will pass through unchanged and not trigger the warning.

----

The reasoning for including this was to be thorough so that in the case where the user accidentally passes an invalid width (or height), the component would resize to default and the user may not understand why.

Sounds like you think this is too much detail to check for.  I am fine with removing the whole thing if you think so.

Reply:
I think, unless the logic differs by type, we shouldn't check types. Eventually we'll have to look into `PropTypes` or `d.ts` to define the prop types. Additionally, fractional values are fine as well, so we don't need to do `Math.floor()`.

Reply:
Sounds good, I have updated this.  There is no type checking now.  It will only do the null coalescing operation (`??`).  We can update the type checking in the future when we get `PropTypes`